### PR TITLE
net/tcp: Add poll_accept, poll_shared_accept and poll_shutdown

### PR DESCRIPTION
**Pull request by dignifiedquire**
### What does this PR do?

Adds 
- `TcpListener::poll_accept`,
- `TcpListener::poll_shared_accept` and
- `TcpStream::poll_shutdown` 

and internally adds `poll_collect_rw`.

### Motivation

The main methods I need right now for #531.

### CODE REVIEW

File: glommio/src/net/tcp_socket.rs [hunk 1]
```diff
@@ -208,6 +233,19 @@ impl TcpListener {
         Ok(a.bind_to_executor())
     }
```
Comment (HippoBaro) [inline glommio/src/net/tcp_socket.rs:234]: I think we should use `poll_fn` here like elsewhere for the sake of consistency

**Review by HippoBaro — APPROVED** [review]
Some small nits, otherwise I am okay to merge

**Comment by pbdeuchler** [comment]
Is this PR dead? I'm in need of a non-blocking way to accept connections on a `TcpListener` and I believe this PR solves my problem

**Comment by vlovich** [comment]
@glommer this seems fine to merge (I don't need it but just working through the backlog). There's an outstanding nit where the poll_shutdown isn't as absolutely concise as it could be (neither is `poll_accept` in my estimation), but either we can merge it or I can push a rebased PR with those nits fixed to get this closed out.

**Pull request merged**